pact-repl/Pact/Core/IR/Eval/Direct/CoreBuiltin.hs

```diff
@@ -1895,10 +1896,23 @@ poseidonHash info b _env = \case
   [VList as]
     | not (V.null as) && length as <= 8,
     Just intArgs <- traverse (preview (_PLiteral . _LInteger)) as -> do
-      chargeGasArgs info (GPoseidonHashHackAChain (length intArgs))
+      chargeGasArgs info (GHashOp (GHashPoseidon (length intArgs)))
       return $ VInteger (poseidon (V.toList intArgs))
   args -> argsError info b args
 
+coreHashKeccak256 :: (IsBuiltin b) => NativeFunction e b i
+coreHashKeccak256 info b _env = \case
+  [VList li] -> do
+    texts <- traverse (asString info b) li
+    let chunkBytes = V.map (BS.length . T.encodeUtf8) texts
+    chargeGasArgs info (GHashOp (GHashKeccak chunkBytes))
+    output <- case keccak256 texts of
```

Comment:
You already compute T.encodeUtf8 of the inputs for gassing, why not re-use that same result and pass that into the native impl, so keccak256 takes a Vector ByteString instead